Debugger disconnects while stepping

#### Issue details
Debugger disconnects from the application while stepping.
Debugger log and application attached.

#### Reproduction steps/code
Open the project in Idea on Mac OS X, set a breakpoint at Main.java:48.
Run on iOS device.
There is a series of System.out.printlns.
Debugger disconnects on 2-3 step through these printlns.

#### Configuration

**Build Tools:**
- [X] IDEA plugin
- [ ] Eclipse plugin
- [ ] Gradle plugin
 
[untitled.zip](https://github.com/MobiVM/robovm/files/3334252/untitled.zip)
[debugger1561629262933.log](https://github.com/MobiVM/robovm/files/3334253/debugger1561629262933.log)
**Versions:**

Robovm: 2.3.7
XCode: 10.2.1
JDK: 1.8.0_202
Mac OS X: Mojave 10.14.5
IntelliJ Idea: Community 2019.1.3

**Build Targets:**
iPhone SE 12.3.1
---

#### Stacktrace
```java
12:54:30.385 SEVERE: Debugger: Thread EventCenterThread crashed
org.robovm.debugger.DebuggerException: THREAD_SUSPENDED: Thread 108fddf30 is not recognized!
	at org.robovm.debugger.delegates.JdwpEventCenterDelegate.getThread(JdwpEventCenterDelegate.java:739)
	at org.robovm.debugger.delegates.JdwpEventCenterDelegate.processSingleEvent(JdwpEventCenterDelegate.java:507)
	at org.robovm.debugger.delegates.JdwpEventCenterDelegate.processEventsCycle(JdwpEventCenterDelegate.java:468)
	at org.robovm.debugger.delegates.JdwpEventCenterDelegate.lambda$onConnectedToTarget$0(JdwpEventCenterDelegate.java:113)
	at java.lang.Thread.run(Thread.java:748)
	at org.robovm.debugger.utils.DebuggerThread.run(DebuggerThread.java:42)
12:54:30.918 INFO: JdwpDebugServer: handling req: ThreadReference(11).FramesCount(7)
12:54:31.396 SEVERE: Debugger: Thread HooksChannel socket thread crashed
org.robovm.debugger.DebuggerException: java.io.IOException: org.robovm.libimobiledevice.LibIMobileDeviceException: IDEVICE_E_UNKNOWN_ERROR
	at org.robovm.debugger.hooks.HooksChannel.doSocketWork(HooksChannel.java:152)
	at org.robovm.debugger.hooks.HooksChannel.lambda$new$0(HooksChannel.java:65)
	at java.lang.Thread.run(Thread.java:748)
	at org.robovm.debugger.utils.DebuggerThread.run(DebuggerThread.java:42)
Caused by: java.io.IOException: org.robovm.libimobiledevice.LibIMobileDeviceException: IDEVICE_E_UNKNOWN_ERROR
	at org.robovm.libimobiledevice.IDeviceConnection$DeviceInputStream.read(IDeviceConnection.java:202)
	at org.robovm.debugger.utils.bytebuffer.ByteBufferPacket.fillFromInputStream(ByteBufferPacket.java:185)
	at org.robovm.debugger.hooks.HooksChannel.doSocketWork(HooksChannel.java:120)
	... 3 more
Caused by: org.robovm.libimobiledevice.LibIMobileDeviceException: IDEVICE_E_UNKNOWN_ERROR
	at org.robovm.libimobiledevice.IDevice.checkResult(IDevice.java:197)
	at org.robovm.libimobiledevice.IDeviceConnection.receive(IDeviceConnection.java:109)
	at org.robovm.libimobiledevice.IDeviceConnection.receive(IDeviceConnection.java:80)
	at org.robovm.libimobiledevice.IDeviceConnection$DeviceInputStream.read(IDeviceConnection.java:200)
	... 5 more
```

--- a/compiler/vm/core/src/hooks.c
+++ b/compiler/vm/core/src/hooks.c
@@ -275,6 +275,11 @@ void _rvmHookThreadStarting(Env* env, Object* threadObj, Thread* thread) {
 
 void _rvmHookThreadDetaching(Env* env, Object* threadObj, Thread* thread, Object* throwable) {
     DEBUGF("Thread %p detaching, threadObj: %p, thread: %p, throwable: %p", rvmRTGetThreadId(env, threadObj), threadObj, thread, throwable);
+    // once got notified that thread is detaching -- ignore any future exceptions or instrumented breakpoints
+    // as JDWP side considers this thread as dead and will fail to lookup it
+    DebugEnv* debugEnv = (DebugEnv*)env;
+    debugEnv->ignoreExceptions = TRUE;
+    debugEnv->ignoreInstrumented = TRUE;
     rvmLockMutex(&writeMutex);
     ChannelError error = { 0 };
     writeChannelByte(clientSocket, EVT_THREAD_DETTACHED, &error);

